src/database/DataTypes/Sensor.js

```diff
@@ -5,9 +5,199 @@
+    return latestAggregatedLog.timestamp;
+  }
+
+  get isInBreach() {
+    const { latestAggregatedLog } = this;
+    if (latestAggregatedLog === null) return false;
+    return latestAggregatedLog.isInBreach;
+  }
+
+  /**
+   * Send a command to the physical device this model
+   * represents, returning the result.
+   * Return object shapes:
+   * {
+   *  success: true/false,
+   *  data,
+   *  error [when success = false]
+   * }
+   */
+  async sendCommand(command) {
+    const { BleManager } = NativeModules;
+    try {
+      const manufacturerId = MANUFACTURER_ID;
+      return await BleManager.sendCommand(manufacturerId, this.macAddress, command);
+    } catch (error) {
+      return { success: false, error };
+    }
+  }
+
+  /**
+   * Sends a blink command, request the LED on the physical
+   * sensor light up.
+   * Return object:
+   * { success: true/false }
+   */
+  async sendBlink() {
+    try {
+      const command = getCommand(MANUFACTURER_ID, 'BLINK');
+      return await this.sendCommand(command);
+    } catch (error) {
+      return error;
+    }
+  }
+
+  /**
+   * Sends reset commands to a physical sensor, resetting the
+   * logging interval, causing all currently stored logs to be
+   * deleted and resetting the sensor advertisement frequency.
+   * Return object:
+   * { success: true/false }
+   */
+  async sendReset() {
+    try {
+      const firstCommand = getCommand(MANUFACTURER_ID, 'RESET_LOG_FREQ');
+      const firstResult = await this.sendCommand(firstCommand);
+      const { success: firstSuccess } = firstResult;
+      if (!firstSuccess) return false;
+
+      const secondCommand = getCommand(MANUFACTURER_ID, 'RESET_ADV_FREQ');
+      const secondResult = await this.sendCommand(secondCommand);
+      const { success: secondSuccess } = secondResult;
+      if (!secondSuccess) return false;
+
+      return true;
+    } catch (error) {
+      return { success: false, error };
+    }
+  }
+
+  /**
+   * Downloads all stored logs from this sensor. Returns an object in the shape
+   * {
+   *   success: true,
+   *   data: [{ logs: [] }]
+   * }
+   * or on not being able to download:
+   * {
+   *  success: false,
+   *  error,
+   * }
+   */
+  async downloadLogs() {
+    try {
+      const command = getCommand(MANUFACTURER_ID, 'DL_LOGS');
+      return await this.sendCommand(command);
+    } catch (error) {
+      return { success: false, error };
+    }
+  }
+
+  /**
+   * Connects with the native bluetooth adapter to scan for this sensor
+   * and retrieve the advertisement data, updating this sensor.
+   * Return object: See BleParser.java
+   */
+  async scanAndUpdate({ manufacturerId, database }) {
+    const { BleManager } = NativeModules;
+    try {
+      const result = await BleManager.getDevices(manufacturerId, this.macAddress);
+      const { success, data } = result;
+      if (!success) return result;
+      if (!data || data.length === 0) return result;
```

Comment:
interesting, checking data here but not in scanForSensors (any reason ?)

Reply:
just because of the operations on the other method and this one not so